lib/fedex/request/rate.rb

```diff
@@ -15,13 +15,15 @@ def process_request
           rate_reply_details = response[:rate_reply][:rate_reply_details] || []
           rate_reply_details = [rate_reply_details] if rate_reply_details.is_a?(Hash)
 
-          rate_reply_details.map do |rate_reply|
+          rates = rate_reply_details.map do |rate_reply|
             rate_details = [rate_reply[:rated_shipment_details]].flatten.first[:shipment_rate_detail]
             rate_details[:service_type] = rate_reply[:service_type]
             rate_details[:transit_time] = rate_reply[:transit_time]
             rate_details[:special_rating_applied] = rate_details[:special_rating_applied]
-            Fedex::Rate.new(rate_details, request_xml: @request_xml, response_xml: @response_xml)
+            Fedex::Rate.new(rate_details)
           end
+
+          Fedex::Rates.new(rates, request_xml: @request_xml, response_xml: @response_xml)
```

Comment:
Unfortunately this introduces a breaking change, which we should avoid unless necessary, it definitely would require a major bump at least. 
Perhaps we could introduce a new a method in https://github.com/veeqo/fedex/blob/74a67b800195ec98313ef39cf2bdb9264d4d820c/lib/fedex/shipment.rb to add the new behaviour or make this return optional via options or ENV vars?